tests/sla_tests.js

```diff
@@ -385,46 +493,505 @@ describe('SLA Bridge and Channels Tester', function() {
+      ' when a call in progress', function(done) {
+    var client = getMockClient();
+    validEndpoints = ['SIP/phone1', 'SIP/phone2', 'SIP/phone3', 'SIP/phone4'];
+
+    var channel = getMockChannel();
+    channel.name = 'SIP/phone3';
+    channel.caller = {'number': '1234'};
+    channel.inbound = true;
+
+    var secondChannel = getMockChannel();
+    secondChannel.name = 'SIP/phone4';
+    secondChannel.caller = {'number': '5678'};
+    secondChannel.inbound = true;
+
+    var sla = require('../lib/sla.js')(client, config, channel, '42')
+      .catch(errHandler)
+      .done();
+
+    setTimeout(function () {
+      var sla2 = require('../lib/sla.js')(client, config, secondChannel, '42')
+      .catch(function (err) {
+        if (err.name === 'ExtensionOccupied') {
+          done();
+        }
+      })
+      .done();
+    }, asyncDelay * 2);
+  });
+
+  it('should test whether or not the application hangs up a non-station ' +
+      'channel when both station channels hang up', function(done) {
+    var client = getMockClient();
+    var channel = getMockChannel();
+    channel.name = 'SIP/phone1';
+    channel.caller = {'number': '1234'};
+    channel.outbound = true;
+
+    var secondChannel;
+    var config = 'tests/testConfigs/multipleEndpoints.json';
+
+    var sla = require('../lib/sla.js')(client, config, channel, '42')
+      .catch(errHandler)
+      .done();
+    var toDial =['1','0','0','#'];
+    var index = 0;
+    var hangupRequested = false;
+
+    bothStationsHangup();
+    setTimeout(function () {
+      secondChannel = getMockChannel();
+      secondChannel.name = 'SIP/phone2';
+      secondChannel.caller = {'number': '5678'};
+      secondChannel.outbound = true;
+      var sla2 = require('../lib/sla.js')(client, config, secondChannel, '42')
+      .catch(errHandler)
+      .done();
+    }, asyncDelay * 7);
+
+    function bothStationsHangup() {
+      setTimeout(function() {
+        if (channels.length === 0 && bridgeChannels.length === 0 &&
+          channel.outbound && channel.wasAnswered && channel.wasHungup &&
+          secondChannel.outbound && secondChannel.wasAnswered && 
+          secondChannel.wasHungup && dialed[0].wasAnswered &&
+          dialed[0].nonStation && dialed[0].wasHungup) {
+          done();
+        } else {
+          if(toDial[index]) {
+            channel.emit('ChannelDtmfReceived', {digit: toDial[index],
+              channel: channel}, {channel: channel});
+            index += 1;
+          }
+          if (secondChannel && channel && bridgeChannels.length === 3 &&
+            !hangupRequested) {
+              hangupRequested = true;
+            channel.hangup(function (){});
+            secondChannel.hangup(function (){});
+          }
+          bothStationsHangup();
+        }
+      }, asyncDelay);
+    } 
+  });
+  it('should test whether the application keeps the non-station channel with ' +
```

Comment:
There should be a blank line between tests here. Overall your pull request could really do with more vertical spacing to help readability.